### PR TITLE
fix xl openmp target backend

### PR DESCRIPTION
**Pull request by trws**
This PR fixes all build and run issues for OpenMP target offload with
the XL compiler as of the Feb 19 version.  Old workarounds removed,
atomic ref tests broken out to parallelize them and reduce amount of
them compiled by default, and mark a destructor declare target that is
not getting implicitly marked for some reason.

### CODE REVIEW

**Comment by trws** [comment]
Turns out this fixes clang upstream 2019-03-19.

File: include/RAJA/policy/openmp_target/reduce.hpp [hunk 1]
```diff
@@ -193,6 +193,9 @@ struct TargetReduce
   {
   }
 
+#ifdef __ibmxl__ // implicit declare target doesn't pick this up
```
Comment (davidbeckingsale) [inline include/RAJA/policy/openmp_target/reduce.hpp]: We don't have a precedent for this, but perhaps this comment would be good with a `TODO:` prefix?

```suggestion
#ifdef __ibmxl__ // TODO: implicit declare target doesn't pick this up
```

File: include/RAJA/policy/openmp_target/reduce.hpp [hunk 2]
```diff
@@ -205,6 +208,9 @@ struct TargetReduce
       }
     }
   }
+#ifdef __ibmxl__ // implicit declare target doesn't pick this up
```
Comment (davidbeckingsale) [inline include/RAJA/policy/openmp_target/reduce.hpp]:
```suggestion
#ifdef __ibmxl__ // TODO: implicit declare target doesn't pick this up
```

Same suggestion as above.

**Comment by davidbeckingsale** [comment]
@trws were you expecting this build failure: https://travis-ci.org/LLNL/RAJA/jobs/526421537#L751 ?

**Comment by trws** [comment]
I was not, but I know what it is.  Will get that cleaned up so we can push this through.

**Comment by rhornung67** [comment]
@trws I updated the base branch on this PR. When it passes tests, should I merge, or do you want the honor?

**Comment by trws** [comment]
Feel free. Thanks!

On 12 Jun 2019, at 14:00, Rich Hornung wrote:

@trws<https://github.com/trws> I updated the base branch on this PR. When it passes tests, should I merge, or do you want the honor?

**Comment by rhornung67** [comment]
@davidbeckingsale do you know what's going on with the codecov stuff? Is this the same thing you saw in Umpire?

**Comment by davidbeckingsale** [comment]
@rhornung67 - nothing, it's setup to report a "fail" if the coverage drops. I think we might be able to change that though.

The issue I fixed was that for the past few months it has not been running at all!

**Pull request merged**